### PR TITLE
feat(Capacity) pay_with_capacity extrinsic

### LINKED ISSUE #1025: [Capacity][SubTask-3-of-4] create extrinsic `pay_with_capacity` in the above pallet, which would wrap the transaction being called (include benchmarks, weights)
This is a subtask of #1010 as the parent see for details

### PR DESCRIPTION
**Pull request by johnmela**
# Goal
The goal of this PR is to add an extrinsic `pay_with_capacity` that will be called as a parent to frequency extrinsics in which capacity may be used as funding.

Closes #1025

### CODE REVIEW

File: pallets/frequency-tx-payment/src/lib.rs [hunk 1]
```diff
@@ -52,11 +52,38 @@ pub mod pallet {
...
+			let dispatch_info = call.get_dispatch_info();
+			(dispatch_info.weight, dispatch_info.class)
+		})]
+		pub fn pay_with_capacity(
+			origin: OriginFor<T>,
+			call: Box<<T as Config>::RuntimeCall>,
+		) -> DispatchResult {
+			ensure_signed(origin.clone())?;
+
+			let _result = call.dispatch(origin);
```
Comment (johnmela) [inline pallets/frequency-tx-payment/src/lib.rs:83]: We may want to emit an event here

**Review by shannonwells — COMMENTED** [review]
We should discuss whether to have a capacity txn event and what it should be called.

File: pallets/frequency-tx-payment/src/lib.rs [hunk 2]
```diff
@@ -52,11 +52,38 @@ pub mod pallet {
...
 	#[pallet::generate_deposit(pub(super) fn deposit_event)]
 	pub enum Event<T: Config> {}
+
+	#[pallet::call]
+	impl<T: Config> Pallet<T> {
+		/// Dispatch the given call as a sub_type of pay_with_capacity. Calls dispatched in this
+		/// fashion, if allowed, will pay with Capacity,
+		#[pallet::call_index(0)]
+		#[pallet::weight({
+			let dispatch_info = call.get_dispatch_info();
```
Comment (enddynayn) [inline pallets/frequency-tx-payment/src/lib.rs:74]: T::WeightInfo::pay_with_capacity().saturating_add(dispatch_info.weight)

**Review by shannonwells — COMMENTED** [review]
@JoeCap08055 's comment about not pulling in other pallets in our mocks unless absolutely necessary made me question whether we need to use the MSA pallet for testing calls.

File: pallets/frequency-tx-payment/src/tests.rs [hunk 3]
```diff
@@ -147,3 +149,44 @@ fn transaction_payment_with_token_and_post_dispatch_refund_is_succesful() {
...
+
+	let pay_with_capacity_call = Box::new(RuntimeCall::FrequencyTxPayment(
+		FrequencyTxPaymentCall::<Test>::pay_with_capacity { call: create_msa_call },
+	));
+	let pay_with_capacity_dispatch_info = pay_with_capacity_call.get_dispatch_info();
+
+	assert_eq!(create_msa_dispatch_info.weight, pay_with_capacity_dispatch_info.weight);
+	// let balance_factor = 10;
+
+	// ExtBuilder::default()
```
Comment (enddynayn) [inline pallets/frequency-tx-payment/src/tests.rs:185]: I think we left some stuff over.

**Review by shannonwells — APPROVED** [review]
shipit

**Review by mattheworris — COMMENTED** [review]
✅

**Review by enddynayn — APPROVED** [review]
👍 amazing!

File: pallets/frequency-tx-payment/src/tests.rs [hunk 4]
```diff
@@ -147,3 +149,44 @@ fn transaction_payment_with_token_and_post_dispatch_refund_is_succesful() {
...
+
+			assert_ok!(FrequencyTxPayment::pay_with_capacity(
+				RuntimeOrigin::signed(signer),
+				create_msa_call
+			));
+		});
+}
+
+#[test]
+fn pay_with_capacity_returns_weight_of_child_call() {
```
[the remark this answers is not in the document]
Comment (shannonwells) [reply pallets/frequency-tx-payment/src/tests.rs:173]: nm, @enddynayn made me realize we are going to need this anyway, at least until we decide to refactor with providers instead

**Pull request merged**